### PR TITLE
[Extended] Add symfony 3.0 support

### LINKED ISSUE #366: Make it compatible with Symfony 3.0
Is it planned to make it compatible this Bundle with Symfony 3.0?

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR is based on #371 and only fixes the tests.
Happy new year!

Closes #371
Closes #366

### CODE REVIEW

**Comment by Ener-Getick** [comment]
@stof @willdurand can you take a look at this please?

ping @luispabon

**Comment by Spomky** [comment]
Looks good to be except that composer.json now requires @dev version of `oauth2-php`.
@alanbem what do you think?

**Comment by Ener-Getick** [comment]
@Spomky this can be reverted once a new version of ``oauth2-php`` is released (only ``master`` is compatible with symfony 3 actually)

File: Form/Handler/AuthorizeFormHandler.php [hunk 1]
```diff
@@ -41,17 +59,25 @@ public function isRejected()
 
     public function process()
     {
-        $this->form->setData(new Authorize(
-            $this->request->request->has('accepted'),
-            $this->request->query->all()
-        ));
+        try {
+            $request = $this->container->get('request_stack')->getCurrentRequest();
```
Comment (merk) [inline Form/Handler/AuthorizeFormHandler.php]: Request stack is not available in 2.3

File: Resources/config/security.xml [hunk 2]
```diff
@@ -18,7 +18,7 @@
         </service>
 
         <service id="fos_oauth_server.security.authentication.listener" class="%fos_oauth_server.security.authentication.listener.class%" public="false">
-            <argument type="service" id="security.context"/>
+            <argument type="service" id="security.token_storage"/>
```
Comment (merk) [inline Resources/config/security.xml]: Token storage not available in 2.3

Comment (merk) [inline Resources/config/security.xml]: Sorry, missed the compiler pass.

File: Form/Handler/AuthorizeFormHandler.php (hunk 1 above)
Comment (merk) [inline Form/Handler/AuthorizeFormHandler.php]: And I failed to read 2 lines down.

**Comment by merk** [comment]
Once there is a stable release of oauth2-php that can be depended apon, :+1:

File: Form/Handler/AuthorizeFormHandler.php (hunk 1 above)
Comment (luispabon) [inline Form/Handler/AuthorizeFormHandler.php]: :+1:

**Comment by luispabon** [comment]
Definitely need oauth2-php released, depending on dev is iffy at best.

**Comment by Ener-Getick** [comment]
@luispabon moreover the bundle won't be completely compatible with sf 3 except if the final user also uses the dev version of ``oauth2-php``...

**Comment by luispabon** [comment]
Yes, indeed. I wonder why a new release hasn't been cut yet.

File: composer.json [hunk 3]
```diff
@@ -17,17 +17,19 @@
     ],
     "require": {
         "php": ">=5.3.3",
-        "friendsofsymfony/oauth2-php": "~1.1.0",
-        "symfony/framework-bundle": "~2.2",
-        "symfony/security-bundle": "~2.1"
+        "friendsofsymfony/oauth2-php": "~1.1",
```
Comment (luispabon) [inline composer.json]: @Ener-Getick seems like you already released oauth2-php 1.2 compatible with symfony/http-foundation 3?

Comment (Ener-Getick) [inline composer.json]: Yeah, I'm just updating several part of the code to not introduce BC breaks.

**Comment by Ener-Getick** [comment]
I'll create ``1.5.0-beta1`` to be sure everything is working as expected.
Feedbacks are welcome :smile:

**Comment by luispabon** [comment]
:+1: 

You joined FOS then?

**Comment by Ener-Getick** [comment]
yeah, see https://github.com/FriendsOfSymfony/friendsofsymfony.github.com/issues/56#issuecomment-171282121

**Comment by luispabon** [comment]
Awesome :+1:

**Comment by Ener-Getick** [comment]
Any idea why travis doesn't launch the builds?

**Comment by luispabon** [comment]
It does, but it takes quite some time for them to clear through the queue. Travis could do with a little more capacity.

**Comment by Ener-Getick** [comment]
Ah yeah just when I post a message they are launched ^^

**Comment by j0k3r** [comment]
@Ener-Getick https://www.traviscistatus.com/incidents/pfcclfpffppn :confused:

**Comment by Ener-Getick** [comment]
@j0k3r ah right thanks! I'll disable ``sudo``

**Comment by Ener-Getick** [comment]
The build failure is related to https://www.traviscistatus.com/incidents/pfcclfpffppn so I merge this PR.

**Pull request merged**